LayoutTests/accessibility/ax-thread-text-apis/text-marker-word-nav.html

```diff
@@ -0,0 +1,191 @@
+    currentMarker = advanceAndVerify(currentMarker, 2, text2);
+    currentMarker = advanceAndVerify(currentMarker, 1, text2);
+
+    // Check user-select:none is also working.
+    var text3 = accessibilityController.accessibleElementById("text3");
+    textMarkerRange = text3.textMarkerRangeForElement(text3);
+    currentMarker = text3.startTextMarkerForTextMarkerRange(textMarkerRange);
+    currentMarker = advanceAndVerify(currentMarker, 1, text3);
+
+    // Check multi-language, Chinese here.
+    var text4 = accessibilityController.accessibleElementById("text4");
+    textMarkerRange = text4.textMarkerRangeForElement(text4);
+    currentMarker = text4.startTextMarkerForTextMarkerRange(textMarkerRange);
+    // Make sure when we are at the beginning of line, it won't go to previous node.
+    // FIXME: Left word in the live tree returns a word, which is inconsistent with other left words when at the beginning of a word.
+    currentMarker = advanceAndVerify(currentMarker, 0, text4);
+    // FIXME: Currently returns an incorrect leftword: 克力 instead of 巧克力. Appears the word break iterator is not grabbing the first character.
+    currentMarker = advanceAndVerify(currentMarker, 2, text4);
+    currentMarker = advanceAndVerify(currentMarker, 1, text4);
+    currentMarker = advanceAndVerify(currentMarker, 1, text4);
+
+    // // And Arabic
+    var text4a = accessibilityController.accessibleElementById("text4a");
+    textMarkerRange = text4a.textMarkerRangeForElement(text4a);
+    currentMarker = text4a.startTextMarkerForTextMarkerRange(textMarkerRange);
+    currentMarker = advanceAndVerify(currentMarker, 3, text4a);
+    currentMarker = advanceAndVerify(currentMarker, 1, text4a);
+
+    // Check text in pre tag with line breaks.
+    var text5 = accessibilityController.accessibleElementById("text5");
+    textMarkerRange = text5.textMarkerRangeForElement(text5);
+    currentMarker = text5.startTextMarkerForTextMarkerRange(textMarkerRange);
+    // At "h" in "both", right word should be "   ".
+    currentMarker = advanceAndVerify(currentMarker, 4, text5);
+    // At the end of first line, right word should be new line.
+    currentMarker = advanceAndVerify(currentMarker, 9, text5);
+
+    // Check text with br tag in it.
+    var text6 = accessibilityController.accessibleElementById("text6");
+    textMarkerRange = text6.textMarkerRangeForElement(text6);
+    currentMarker = text6.startTextMarkerForTextMarkerRange(textMarkerRange);
+    currentMarker = advanceAndVerify(currentMarker, 4, text6); /* Right word here is *linebreak* */
+    // FIXME: Previous expectation had left and right word both be empty. This new expectation expects that the right and left words are the words on either side of the line break.
+    currentMarker = advanceAndVerify(currentMarker, 1, text6);
+
+    // FIXME: Skipping text 8 from the original text (which tests <audio> on the live tree), due to invalidation issues with <audio controls> and the AXObjectCache.
```

Comment:
I'm curious (and would be nice to include in the comment if you remember) — what are the details of this invalidation issue? Is it an ITM cache invalidation issue?

Reply:
This was due to `gAccessibilityThreadTextApis` being set to the wrong value by another document. This was fixed with https://github.com/WebKit/WebKit/commit/18aeb0e661445201b72923ce20216bc52033b2a3